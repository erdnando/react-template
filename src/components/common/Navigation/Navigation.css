.navigation {
  background-color: #343a40;
  padding: 1rem;
}

.nav-list {
  list-style: none;
  display: flex;
  gap: 1rem;
  margin: 0;
  padding: 0;
}

.nav-item {
  display: flex;
}

.nav-link {
  color: #fff;
  text-decoration: none;
  padding: 0.5rem 1rem;
  border-radius: 4px;
  transition: background-color 0.3s;
}

.nav-link:hover {
  background-color: #495057;
}

.nav-link.active {
  background-color: #007bff;
}
