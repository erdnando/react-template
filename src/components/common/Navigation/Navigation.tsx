import React from 'react';
import { Link } from 'react-router-dom';
import './Navigation.css';

const Navigation: React.FC = () => {
    return (
        <nav className="navigation">
            <ul>
                <li>
                    <Link to="/">Home</Link>
                </li>
                <li>
                    <Link to="/catalogs">Catalogs</Link>
                </li>
                <li>
                    <Link to="/users">Users</Link>
                </li>
                <li>
                    <Link to="/permissions">Permisos</Link>
                </li>
                <li>
                    <Link to="/login">Login</Link>
                </li>
                <li>
                    <Link to="/reset-password">Reset Password</Link>
                </li>
                <li>
                    <Link to="/sign-out">Sign Out</Link>
                </li>
            </ul>
        </nav>
    );
};

export default Navigation;