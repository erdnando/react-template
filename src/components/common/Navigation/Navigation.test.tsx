import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navigation from './Navigation';

const renderWithRouter = (component: React.ReactElement) => {
  return render(
    <BrowserRouter>
      {component}
    </BrowserRouter>
  );
};

describe('Navigation Component', () => {
  test('renders navigation links', () => {
    renderWithRouter(<Navigation />);
    const homeLink = screen.getByText(/home/i);
    const catalogLink = screen.getByText(/catalogs/i);
    const usersLink = screen.getByText(/users/i);
    expect(homeLink).toBeInTheDocument();
    expect(catalogLink).toBeInTheDocument();
    expect(usersLink).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = renderWithRouter(<Navigation />);
    expect(asFragment()).toMatchSnapshot();
  });
});