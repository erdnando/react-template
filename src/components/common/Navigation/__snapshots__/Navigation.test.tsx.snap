// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navigation"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/catalogs"
        >
          Catalogs
        </a>
      </li>
      <li>
        <a
          href="/users"
        >
          Users
        </a>
      </li>
      <li>
        <a
          href="/login"
        >
          Login
        </a>
      </li>
      <li>
        <a
          href="/reset-password"
        >
          Reset Password
        </a>
      </li>
      <li>
        <a
          href="/sign-out"
        >
          Sign Out
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
